### PR TITLE
k8s: Replaces content in Install pod network add-on w symlinks.

### LINKED ISSUE #392: Kubernetes: inconsistent and incorrect CNI dir
**Environment (please complete the following):**
 - Clear Linux OS version: 27670
 - https://clearlinux.org/documentation/clear-linux/tutorials/kubernetes as of 2019-02-08

Handling of the CNI directory is wrong in the current revision of the page.

```
CRI-O’s default plugin_dir is /opt/bin/cni. CNI plugins are installed as part of cloud-native-basic.
```

The actual path is `/opt/cni/bin`.

```
Notes about Weave Net add-on
...
Add the loopback CNI plugin to the plugin path with the command:

sudo ln -s /usr/libexec/cni/loopback /opt/bin/cni/loopback
```

`/usr/lib/tmpfiles.d/cri-o.conf` creates `/opt/bin/cni`, so the command above doesn't fail, but probably doesn't fix the issue (haven't tried).

```
Notes about flannel add-on
...
Create a symlink for the network overlays:

sudo ln -s /usr/libexec/cni /opt/cni/bin
```

This will fail once `/usr/lib/tmpfiles.d/cri-o.conf`gets changed to create the directory.

```
Note

Clear Linux OS installs CNI plugins that are part of the cloud-native-basic bundle to /usr/libexec/cni. The directory is required because ` swupd verify` may use it to repair a system to a known good state.
```

That's correct, but it is not specific to flannel.

My proposal is to have one note that works for multiple CNI plugins and remove the workarounds from the flannel and weave sections:

```


Note

CRI-O’s default plugin_dir is /opt/cni/bin. This must be a writable directory, because third-party networking add-ons will install themselves there. CNI plugins provided by Clear Linux are installed as part of cloud-native-basic in /usr/libexec/cni/ and are currently *not* found by CRI-O by default. These separate directories are required because `swupd` controls the content of `/usr` and leaves `/opt` unchanged.

When using third-party network add-ons that rely on those plugins (like weave or flannel do), then make them available via symlinks:

sudo mkdir -p /opt/cni/bin
for i in /usr/libexec/cni/*; do sudo ln -sf $i /opt/bin/cni/; done
```

### PR DESCRIPTION
**Pull request by mvincerx**
- Provides bash script to manage default plugin_dir
- Closes #392

Signed-off-by: Michael Vincerra <michael.vincerra@intel.com>

### CODE REVIEW

**Comment by mvincerx** [comment]
@pohly, Please review asap. 
@DnPlas, @eadamsintel, Could one of you also review and approve this asap? 
@mythi, FYI only.

File: source/clear-linux/tutorials/kubernetes.rst [hunk 1]
```diff
@@ -158,54 +158,34 @@ Configure and run CRI-O + kata-runtime
...
-For using CRI-O and ``Weave Net``, complete the following step.
+.. code-block:: bash
 
-Add the `loopback` CNI plugin to the plugin path with the command:
+   sudo mkdir -p /opt/cni/bin
 
 .. code-block:: bash
 
-   sudo ln -s /usr/libexec/cni/loopback /opt/bin/cni/loopback
+   for i in /usr/libexec/cni/*; do sudo ln -sf $i /opt/bin/cni/; done
```
Comment (mythi) [inline source/clear-linux/tutorials/kubernetes.rst]: should be ```/opt/cni/bin```! :)

Comment (mvincerx) [reply source/clear-linux/tutorials/kubernetes.rst]: See latest commit please. Please verify. [Sorry but I don't have time to validate this on a test box.]

File: source/clear-linux/tutorials/kubernetes.rst [hunk 2]
```diff
@@ -150,62 +150,41 @@ Configure and run CRI-O + kata-runtime
...
-..  code-block:: bash
-
-    --pod-network-cidr 10.244.0.0/16
-
-If you are using CRI-O and `flannel` and you want to use Kata Containers, edit the :file:`/etc/crio/crio.conf` file to add:
-
-..  code-block:: bash
-
-    [crio.runtime]
-    manage_network_ns_lifecycle = true
```
Comment (pohly) [inline source/clear-linux/tutorials/kubernetes.rst]: The instructions above for Flannel are still needed. I would keep them.

What I meant is that the instructions regarding /opt/cni/bin can be removed from the Flannel section, not that the Flannel section itself can be removed.

File: source/clear-linux/tutorials/kubernetes.rst [hunk 3]
```diff
@@ -150,62 +150,41 @@ Configure and run CRI-O + kata-runtime
...
-**Notes about Weave Net add-on**
+   CNI plugins provided by |CL| are installed as part of *cloud-native-basic*
+   in :file:`/usr/libexec/cni/` and are currently *not* found by CRI-O by
+   default. These separate directories are required because `swupd` controls
+   the content of :file:`/usr` and leaves :file:`/opt` unchanged.
 
-If you choose the `Weave Net` add-on, you must make the following
-changes because it installs itself in the :file:`/opt/cni/bin` directory.
+When using third-party network add-ons that rely on those plugins, such as
+Weave or Flannel do, make them available by creating symlinks:
```
Comment (pohly) [inline source/clear-linux/tutorials/kubernetes.rst]: For Weave, no specific steps are needed besides the generic setup of /opt/cni/bin. Perhaps it's worth mentioning that? There's a difference between "we don't have instructions for xyz" and "xyz just works".

File: source/clear-linux/tutorials/kubernetes.rst (hunk 2 above)
Comment (mvincerx) [reply source/clear-linux/tutorials/kubernetes.rst]: Review https://github.com/clearlinux/clear-linux-documentation/pull/393/commits/f67cf636d71687b0d02d1a371a6049af440021ea

File: source/clear-linux/tutorials/kubernetes.rst (hunk 3 above)
Comment (mvincerx) [reply source/clear-linux/tutorials/kubernetes.rst]: Review https://github.com/clearlinux/clear-linux-documentation/pull/393/commits/f67cf636d71687b0d02d1a371a6049af440021ea

File: source/clear-linux/tutorials/kubernetes/kubernetes.rst [hunk 4]
```diff
@@ -1,7 +1,12 @@
 .. _kubernetes:
 
+<<<<<<< HEAD:source/clear-linux/tutorials/kubernetes.rst
+Run Kubernetes\*
+################
+=======
```
Comment (pohly) [inline source/clear-linux/tutorials/kubernetes/kubernetes.rst]: Merge conflicts?!

Comment (mvincerx) [reply source/clear-linux/tutorials/kubernetes/kubernetes.rst]: No.  Don't worry. I managed the merge conflicts in a separate commit.  Please review content of the commit only.

File: source/clear-linux/tutorials/kubernetes/kubernetes.rst [hunk 5]
```diff
@@ -150,62 +155,62 @@ Configure and run CRI-O + kata-runtime
...
-`kubeadm init` command:
-
-..  code-block:: bash
-
-    --pod-network-cidr 10.244.0.0/16
-
-If you are using CRI-O and `flannel` and you want to use Kata Containers, edit the :file:`/etc/crio/crio.conf` file to add:
+The CRI-O default plugin_dir is :file:`/opt/cni/bin`. This must be a
+writable directory because third-party networking add-ons will install
+themselves there.
```
Comment (pohly) [inline source/clear-linux/tutorials/kubernetes/kubernetes.rst]: Now you have this "The CRI-O default plugin_dir is..." note twice in the text.

File: source/clear-linux/tutorials/kubernetes/kubernetes.rst (hunk 4 above)
Comment (pohly) [reply source/clear-linux/tutorials/kubernetes/kubernetes.rst]: It still doesn't make sense to me to have commits with merge commits in a PR. But if this is how you want to maintain it, so be it.

**Comment by mvincerx** [comment]
@mythi, Can you please provide any final feedback by Tuesday 12 Feb.?

**Comment by mythi** [comment]
@mvincerx looks good to me! Thanks @pohly too

**Comment by mvincerx** [comment]
Thanks, all.

**Pull request merged**